File: Jack.Presentation/Controllers/API/StatusController.vb
Imports System.Net
Imports System.Web.Http

Namespace Controllers.API
    Public Class StatusController
        Inherits ApiController

        <HttpGet>
        Public Function LoadForChildrem() As IList(Of Model.Status)

            Dim lstRetorno As List(Of Model.Status) = Nothing
            Dim oBusiness As Business.Status = Nothing
            Try
                oBusiness = New Business.Status()
                lstRetorno = oBusiness.LoadForCriancas()

            Catch ex As Exception
                lstRetorno = Nothing
            Finally
                oBusiness = Nothing
            End Try

            Return lstRetorno


        End Function

        <HttpGet>
        Public Function LoadForFamily() As IList(Of Model.Status)

            Dim lstRetorno As List(Of Model.Status) = Nothing
            Dim oBusiness As Business.Status = Nothing
            Try
                oBusiness = New Business.Status()
                lstRetorno = oBusiness.LoadForFamilia()

            Catch ex As Exception
                lstRetorno = Nothing
            Finally
                oBusiness = Nothing
            End Try

            Return lstRetorno

        End Function

    End Class
End Namespace

File: Jack.Presentation.Web/Controllers/API/StatusController.cs

using System;
using System.Collections.Generic;
using System.Web.Http;
using Business = Jack.Business;
using Model = Jack.Model;

namespace Controllers.API
{
	public class StatusController : ApiController
	{
        [HttpGet()]
        public IList<Model.Status> GetValues()
        {

            IList<Model.Status> lstRetorno = null;
            Business.Status oBusiness = null;
            try
            {
                oBusiness = new Business.Status();
                lstRetorno = oBusiness.LoadAll();

            }
            catch (Exception ex)
            {
                lstRetorno = null;
                throw ex;
            }
            finally
            {
                oBusiness = null;
            }

            return lstRetorno;

        }

  //      [HttpGet()]
		//public IList<Model.Status> LoadForChildrem()
		//{

		//	IList<Model.Status> lstRetorno = null;
		//	Business.Status oBusiness = null;
  //          try
  //          {
  //              oBusiness = new Business.Status();
  //              lstRetorno = oBusiness.LoadForCriancas();

  //          }
  //          catch (Exception ex)
  //          {
  //              lstRetorno = null;
  //              throw ex;
  //          }
  //          finally
  //          {
  //              oBusiness = null;
  //          }

		//	return lstRetorno;

		//}

		//[HttpGet()]
		//public IList<Model.Status> LoadForFamily()
		//{

		//	IList<Model.Status> lstRetorno = null;
		//	Business.Status oBusiness = null;
		//	try
  //          {
		//		oBusiness = new Business.Status();
		//		lstRetorno = oBusiness.LoadForFamilia();

		//	}
  //          catch (Exception ex)
  //          {
		//		lstRetorno = null;
  //              throw ex;
  //          }
  //          finally
  //          {
		//		oBusiness = null;
		//	}

		//	return lstRetorno;

		//}

	}
}

